docs/tutorials/HowToCreateAProduct.md

```diff
@@ -0,0 +1,92 @@
+---
+uid: HowToCreateAProduct
+---
+# How to create a resource
+
+This tutorial shows how [Products](xref:Moryx.AbstractionLayer.Products.ProductType) should be implemented. Look [here](../articles/Products/Concept.md) if you are not firm with `Product`. 
+
+For products we differentiate between [ProductType](xref:Moryx.AbstractionLayer.Products.ProductType) and [ProductInstance](xref:Moryx.AbstractionLayer.Products.ProductInstance). The `ProductType` is what you can order in a catalog, while the `ProductInstance` is what you received after ordering: an instance of the product with its unique serialnumber. So that a `ProductType` can be producted it needs a corresponding `ProductInstance`. If your application isn't used for production, you can skip the `ProductInstances`. 
+
+## Create a basic ProductType and ProductInstance
+All ProductTypes are derived from `ProductType`. The methode `Instantiate()` returns an object of the correspoding `ProductInstance`.
+
+With the attibute `DisplayName` you can change the name displayed on the UI. Using `Description` adds descriptions to a property on the UI. All public properties of a `ProductType` will be displayed on the UI.
+```C#
+[DisplayName("Watch Needle")]
+public class NeedleType : ProductType
+{
+    // Properties of the ProductType
+    [Description("Length of the needle")]
+    public int Length { get; set; }
+
+    // Creates the corresponding instance of the productType
+    protected override ProductInstance Instantiate()
+    {
+        return new NeedleInstance();
+    }
+}
+```
+All ProductInstances are derived from `ProductInstance<TProductType>`. Instance specific properties can be added. If you have for example only one materialnumber for a car type and the configuration depends on the user, this could be part of the properties of the instance.
+```C#
+public class NeedleInstance : ProductInstance<NeedleType>
+{
+    // Instance specific properties
+    public NeedleRole Role { get; set; }
+
+    ...
+}
+```
+
+## Use PartLinks
+Some `ProductTypes` have components. These can be modeled using [PartLinks](xref:Moryx.AbstractionLayer.Products.ProductPartLink). If no additional properties are needed, use the basic `ProductPartLink`. This is done for the part *WatchFace* in the example.
+
+```C#
+[DisplayName("Watch")]
+public class WatchType : ProductType
+{
+    // References to product
+    [DisplayName("Watch face")]
+    public ProductPartLink<WatchFaceType> WatchFace { get; set; }
+
+    // References to product using costum part link
+    [DisplayName("Watch needle")]
+    public List<NeedlePartLink> Needles { get; set; } = new List<NeedlePartLink>();
+
+    protected override ProductInstance Instantiate()
+    {
+        return new WatchInstance
+        {
+            //All parts need to be instantiated
+            WatchFace = (WatchFaceInstance)WatchFace.Instantiate(),
+            Needles = Needles.Instantiate<NeedleInstance>()
```

Comment:
would the instantiate of the needleType create a list in this case? 🤔

Reply:
The PartlinkExtension instantiates every NeedleType in Needles and adds them to a list

Reply:
Ahh I didn't see the `Needles` property, now it makes sense 🙈